searchlib/src/vespa/searchlib/btree/btree_key_data.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <cstdint>
+
+namespace search::btree {
+
+/**
+ * Empty class to use as DataT template parameter for BTree classes to
+ * indicate that leaf nodes have no data (similar to std::set having less
+ * information than std::map).  Use of this class triggers the below
+ * partial specialization of BTreeNodeDataWrap to prevent unneeded
```

Comment:
The last sentence refers to code in old location. Should be moved back to BTreeNodeDataWrap and adapted as needed.